### ISSUE TITLE
Unit of published observations

### ISSUE THREAD

**Issue by mbradley612** [matthewwall/weewx-mqtt#5]
Hello Matthew. Thank you for this excellent extension, we are now using at www.hillheadsc.org.uk to provide our live weather station feed.

My configuration is below. You'll see that we are using knots for windSpeed and windGust. From reading the wiki article at https://github.com/weewx/weewx/wiki/mqtt, I was expecting the windSpeed observation to be published as windSpeed_knot.

Instead, when I run mosquitto_sub (on a simulator setup) I get:

```
{"pressure_mbar": "1053.16426107", "windSpeed_kph": "0.00031515680943", "windGust_kph": "0.000378188171316"}
{"pressure_mbar": "1053.16422385", "windSpeed_kph": "0.000319931786004", "windGust_kph": "0.000383918143205"}

```

I'm fairly confident that the conversion is working; if I change the units to an arbitary string, I get an error, and the values I get on our live system appear to be correct.

Is this expected behaviour? Is there something wrong with my config?

Many thanks,

Matthew

```
[[MQTT]]
        topic = weather
        unit_system = METRIC
        binding = loop, archive
        aggregation = aggregate
        obs_to_upload = none
        log_success = False
        server_url = mqtt://localhost:1883
        [[[inputs]]]
            [[[[windSpeed]]]]
                units = knot    # use knots instead of meter_per_second
            [[[[windGust]]]]
                units = knot
            [[[[pressure]]]]
            [[[[windDir]]]]
                format = directions
```

**Comment by mbradley612** [comment]
Hi there,

We think this patch fixes the issue. The patch appends the correct unit label when the units have been specified in the config.

Many thanks,
Matthew

`---
 bin/user/mqtt.py | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/bin/user/mqtt.py b/bin/user/mqtt.py
index 7e16696..296e03e 100644
--- a/bin/user/mqtt.py
+++ b/bin/user/mqtt.py
@@ -169,6 +169,8 @@ UNIT_REDUCTIONS = {
     'km_per_hour2': 'kph',
     'meter_per_second': 'mps',
     'meter_per_second2': 'mps',
+    'knot': 'knot',
+    'knot2': 'knot',
     'degree_compass': None,
     'watt_per_meter_squared': 'Wpm2',
     'uv_index': None,
@@ -177,15 +179,17 @@ UNIT_REDUCTIONS = {
     }
 
 # return the units label for an observation
-def _get_units_label(obs, unit_system):
-    (unit_type, _) = weewx.units.getStandardUnitType(unit_system, obs)
+def _get_units_label(obs, unit_system, unit_type=None):
+    if not unit_type:
+        (unit_type, _) = weewx.units.getStandardUnitType(unit_system, obs)
     return UNIT_REDUCTIONS.get(unit_type, unit_type)
 
 # get the template for an observation based on the observation key
 def _get_template(obs_key, overrides, append_units_label, unit_system):
     tmpl_dict = dict()
     if append_units_label:
-        label = _get_units_label(obs_key, unit_system)
+        unit_type = overrides.get('units')
+        label = _get_units_label(obs_key, unit_system, unit_type)
         if label is not None:
             tmpl_dict['name'] = "%s_%s" % (obs_key, label)
     for x in ['name', 'format', 'units']:
-- 
2.7.4`

**Comment by matthewwall** [comment]
fixed at commit 00e23e8

thank you for the bug report, and even more thanks for the fix!

**Comment by mbradley612** [comment]
The fix was from my 16 year old son, his first public contribution. He wasn't brave enough to submit it himself, after your positive feedback I think he will be next time!
